src/main/java/org/ambraproject/rhino/view/article/RelationshipViewFactory.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (c) 2019 Public Library of Science
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+package org.ambraproject.rhino.view.article;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import com.google.common.collect.ImmutableMap;
+
+import org.ambraproject.rhino.identity.ArticleIdentifier;
+import org.ambraproject.rhino.model.Article;
+import org.ambraproject.rhino.model.ArticleIngestion;
+import org.ambraproject.rhino.model.ArticleRelationship;
+import org.ambraproject.rhino.model.ArticleRevision;
+import org.ambraproject.rhino.service.ArticleCrudService;
+import org.ambraproject.rhino.view.journal.JournalOutputView;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class RelationshipViewFactory {
+  @Autowired
+  private ArticleCrudService articleCrudService;
+
+  public static Map<String, String> invertedTypes = ImmutableMap.<String, String>builder()
+    .put("commentary", "commentary-article")
+    .put("commentary-article", "commentary")
+    .put("companion", "companion")
+    .put("corrected-article", "correction-forward")
+    .put("correction-forward", "corrected-article")
+    .put("retracted-article", "retraction-forward")
+    .put("retraction-forward", "retracted-article")
+    .put("object-of-concern", "concern-forward")
+    .put("concern-forward", "object-of-concern")
+    .put("updated-article", "update-forward")
+    .put("update-forward", "updated-article")
+    .build();
+
+  /* 
+   * Our corpus contains some dirty data: make it canonical.
+   */
+  private static String getCanonicalType(String type) {
+    if (type.equals("corrrection-forward")) {
+      /* lol, it's too much trouble to repub this */
```

Comment:
This data comes from the database via https://github.com/PLOS/rhino/blob/d38cbbe488bd25fee4e2576b94d0b70df8e280d9/src/main/java/org/ambraproject/rhino/service/impl/ArticleCrudServiceImpl.java#L302

If we were to update the database to correct this while leaving the XML alone, would that require a repub?

Reply:
No, it wouldn't require a repub.

I considered something similar, namely doing a canonicalization on ingest and creating a migration to fix the existing data. 

In my opinion, the correct solution is to fix the XML. I would like to push towards a lighter "repub" process that could involve making corrections like this to the XML without the full weight of a repub.

But, since we can't do that correct solution now, I think this offers the easiest solution. It isn't ideal, because it would be to push the fix to the XML. It would IMO be a more correct solution to fix the ingest & the existing data in the db, but, on the other hand, this fix is lighter weight. I'm not wedded to this solution, but it's what i went with.

Reply:
ok, gotcha

Reply:
Thanks for being understanding